docs: document declarative tablespaces (#3414)
cloudnative-pg/cloudnative-pg: docs/src/tablespaces.md

>   different partition and used until the system can be reconfigured.
> - Second, tablespaces allow an administrator to use knowledge of the usage
>   pattern of database objects to optimize performance.

## Declarative tablespaces

CloudNativePG provides support for PostgreSQL tablespaces through **declarative
tablespaces**, operating at two distinct levels:

- Kubernetes: managing persistent volume claims, identically to how PGDATA and
  WAL volumes are handled
- PostgreSQL: managing the `TABLESPACE` global objects within the PostgreSQL
  instance

Being a part of the Kubernetes ecosystem, CloudNativePG's declarative
tablespaces are implemented leveraging Persistent Volume Claims (and Persistent
Volumes). Each tablespace defined in the cluster is housed in its own
persistent volume. CloudNativePG takes care of generating the PVCs, mounting
the required volumes in the instance Pods in normalized locations, and ensuring
replicas are ready to support tablespaces before activating them in the
primary.

Tablespaces can be setup when the cluster is created, or added at a later time
— provided the storage is available when requested. Currently, they cannot be

> removed, but this feature will be introduced in a future minor/patch version
> of CloudNativePG.

Author: ```suggestion
removed, but support for removal will be introduced in a future minor/patch version
of CloudNativePG.
```

  1663 | pg_default
  1664 | pg_global
 16387 | tbs1
 16388 | tbs2
 16389 | tbs3
(5 rows)
```

You can start using them right away:

``` txt
app=# CREATE TABLE fibonacci(num INTEGER) TABLESPACE tbs1;
CREATE TABLE
```

The cluster status has a section for tablespaces:

``` yaml
status:

  <- snipped ->

  tablespacesStatus:
  - name: tbs1
    status: reconciled
  - name: tbs2
    status: reconciled
  - name: tbs3
    status: reconciled
```

## Storage classes and tablespaces

As is the case with PGDATA and WAL volumes, you can use different storage
classes for your tablespaces too. This is a very convenient way of optimizing
your resources, balancing performance and costs of your storage based on data
access usage and expectations.

Let's use the following example to explain the feature:

```yaml
apiVersion: postgresql.cnpg.io/v1
kind: Cluster
metadata:
  name: yardbirds
spec:
  instances: 3

  storage:
    size: 10Gi
  walStorage:
    size: 10Gi
  tablespaces:
    - name: current
      size: 100Gi
      storageClass: fastest
    - name: this_year
      size: 500Gi
      storageClass: balanced
```

The `yardbirds` cluster example above requests 5 persistent volume claims using

> 3 different storage classes:

Reviewer: fixed
